nav2_behavior_tree/plugins/action/planner_selector_node.cpp

```diff
@@ -36,16 +36,25 @@ PlannerSelector::PlannerSelector(
 
   getInput("topic_name", topic_name_);
 
+  rclcpp::QoS qos(rclcpp::KeepLast(1));
+  qos.transient_local().reliable();
+
   planner_selector_sub_ = node_->create_subscription<std_msgs::msg::String>(
-    topic_name_, 1, std::bind(&PlannerSelector::callbackPlannerSelect, this, _1));
+    topic_name_, qos, std::bind(&PlannerSelector::callbackPlannerSelect, this, _1));
 }
 
 BT::NodeStatus PlannerSelector::tick()
 {
   rclcpp::spin_some(node_);
 
   if (last_selected_planner_.empty()) {
-    getInput("default_planner", last_selected_planner_);
+    std::string default_planner;
```

Comment:
A comment here about what this is implementing would be great. Something just letting people know its so that if no default is set & none received failing. If you want to proceed without a set, must set a default. Would also be good context to add to the configuration page for this variable.